src/features/posts.ts

```diff
@@ -0,0 +1,45 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { User } from '../types/User';
+import { getUserPosts } from '../api/posts';
+
+type PostsState = {
+  posts: Post[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: PostsState = {
+  posts: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const getPosts = createAsyncThunk<Post[], User['id']>(
+  'posts/fetch',
+  userId => getUserPosts(userId),
+);
+
+export const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder
+      .addCase(getPosts.fulfilled, (state, action) => {
+        return {
+          ...state,
+          posts: action.payload,
+        };
```

Comment:
When the request is fulfilled, you should also set `hasError` to false. This is because if a previous request failed but a subsequent request succeeds, you want to clear the error state.